homeassistant/components/emoncms/sensor.py

```diff
@@ -150,12 +150,13 @@ def __init__(
             id_for_name = "" if str(sensorid) == "1" else sensorid
             # Use the feed name assigned in EmonCMS or fall back to the feed ID
             feed_name = elem.get("name") or f"Feed {elem['id']}"
-            self._name = f"EmonCMS{id_for_name} {feed_name}"
+            self._attr_name = f"EmonCMS{id_for_name} {feed_name}"
         else:
-            self._name = name
-        self._identifier = get_id(
+            self._attr_name = name
+        self._attr_unique_id = get_id(
             sensorid, elem["tag"], elem["name"], elem["id"], elem["userid"]
         )
```

Comment:
Names cannot be part of a unique identifier.

See also our developer documentation around this, which lists the requirements: <https://developers.home-assistant.io/docs/entity_registry_index/#unique-id>

../Frenck

Reply:
tag and name removed from the unique id

emoncms API does not provide mac address at the moment, so I've kept sensor id, feed id and emoncms user_id

https://github.com/home-assistant/core/pull/117911/commits/a72cc6e41848df68a29dde48a2bd1d5509e4f072

Reply:
Are these unique across different installations?

Reply:
sensor id must be unique as specified in the doc : 
```
id integer Required

Positive integer identifier for the sensor. Must be unique if you specify multiple Emoncms sensors.
```
cf https://www.home-assistant.io/integrations/emoncms/#id

unless I'm mistaken, I understand it is what you call `unique id of last resort` in the doc ?
https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-of-last-resort

I've seen it used like that in the modbus integration

https://github.com/home-assistant/core/blob/d33068d00cc921803af3eaec105d7d15e7c45231/homeassistant/components/modbus/binary_sensor.py#L152

Reply:
> unless I'm mistaken, I understand it is what you call unique id of last resort in the doc ?

No, that is not the ID @ emoncms, that is the UUID of the config flow entry. As this integration doesn't have a configuration flow, it means there is no unique ID of last resort available.

In that case, this integration first needs to be migrated to a config flow (and thus UI configuration).

../Frenck

Reply:
OK sorry, I was misled when I saw that the modbus did that way without having a flow config.

So going to close that PR, there is more work than expected